lms/djangoapps/certificates/views/webview.py

```diff
@@ -249,22 +249,29 @@ def _update_course_context(request, context, course, course_key, platform_name):
                                                               '{partner_short_name}.').format(
             partner_short_name=context['organization_short_name'],
             platform_name=platform_name)
+
+
+def _update_course_run_context(context, course_key):
+    """
+    Updates context dictionary with course run info from Discovery.
+    """
     course_certificate_settings = CertificateGenerationCourseSetting.get(course_key)
-    if course_certificate_settings and course_certificate_settings.language_specific_templates_enabled:
-        fields = ['content_language']
+    if course_certificate_settings:
+        course_run_fields = []
+        if course_certificate_settings.language_specific_templates_enabled:
+            course_run_fields.append('content_language')
         if course_certificate_settings.include_hours_of_effort:
-            fields.extend(['start', 'end', 'max_effort'])
-        course_run_data = get_course_run_details(course_key, fields)
-        if course_run_data.get('start') and course_run_data.get('end') and course_run_data.get('max_effort'):
-            # Calculate duration of the course run in weeks, multiplied by max_effort for total Hours of Effort
-            try:
-                start = parser.parse(course_run_data.get('start'))
-                end = parser.parse(course_run_data.get('end'))
-                max_effort = int(course_run_data.get('max_effort'))
-                context['hours_of_effort'] = ((end - start).days / 7) * max_effort
-            except ValueError:
-                log.exception('Error occurred while parsing course run details')
-        context['content_language'] = course_run_data.get('content_language')
+            course_run_fields.extend(['weeks_to_complete', 'max_effort'])
```

Comment:
is `weeks_to_complete` always present? Is it any more/less reliable than `start` and `end`? Should we fall back to `start` and `end` if it's not available?

Reply:
I'm glad you caught that. I believe it is possible that it might not be included. Falling back to 'start' and 'end' would probably only be best for instructor-paced courses though, but I can include that in the fields that we pull.

Reply:
Hmm, good point about start/end. Maybe we should make weeks to complete a requirement for showing hours of effort then? You could add a note to the help text of the boolean that hours of effort will only be shown if the course has a configured weeks_to_complete and max_effort fields.